File: C#/2_Selection Sort/Selection Sort.cs
using System;

namespace Algorithm {
    
    class SelectionSort {

        public static void selectionSort(int[] arr){
            for(int k = 0; k < arr.Length - 1; k++){
                int min = k;
                for(int j = k + 1; j < arr.Length; j++){
                    if(arr[j] < arr[min]){
                        min = j;
                    }
                }
                int temp = arr[min];
                arr[min] = arr[k];
                arr[k] = temp;
            }
        }

        public static void printArray(int[] arr){
            for(int i = 0; i < arr.Length; i++){
                Console.Write(arr[i] + " ");
            }
        }

        public static void Main(string[] args){
            int[] arr = {64, 25, 12, 22, 11};
            selectionSort(arr);
            printArray(arr);
        }
    }
}

File: Ruby/2_Selection Sort/Selection Sort.rb
def selectionSort(arr)
	for k in 0...(arr.length - 1)
		min = k
		for j in (k + 1)...arr.length
			min = j if arr[j] < arr[min]
		end
		arr[k], arr[min] = arr[min], arr[k]
	end
end

nums = [3, 2, 1, 5, 8, 6, 9, 4, 7]
selectionSort(nums)
p nums